### PR TITLE
feat: add day 6

### PR DESCRIPTION
**Pull request by josejorgers**

### CODE REVIEW

File: tutorial/guias/Paso_a_paso_DIA_6.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+## La clase `Game`
+
+Lo primero, como siempre, es el método `__init__`. En este caso no recibiremos ningún parámetro. En la primera línea iniciamos el motor gráfico de Pygame, que es el que nos muestra todo en pantalla. Luego con `pygame.display.set_caption("Snake")` estamos creando el título de nuestro juego, el texto **Snake** se verá en el borde superior de la ventana del juego. A continuación damos valores a la velocidad de juego, el ancho y la altura de la ventana de juego. El campo `game_window` se crea haciendo `self.game_window = pygame.display.set_mode((self.window_x, self.window_y))`. Este representa la ventana donde se desarrolla el juego y es el que tanto se usa en las funciones que ya vimos el día anterior en la clase `Renderer`. Además creamos el campo `fps` (fraps per second) o sea, la cantidad de veces que se dibuja nuestro juego en la pantalla por segundo. Este valor se toma directamente del reloj de Pygame haciendo `self.fps = pygame.time.Clock()`.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_6.md]: It should be `frames per second`, right?

File: tutorial/guias/Paso_a_paso_DIA_6.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    return any(self.snake.position == block for block in self.snake.body[1:])
+```
+
+Hay varios detalles en esta función. Analicemos primero lo que tenemos dentro de los paréntesis en la sentencia **return**:
+
+```python
+self.snake.position == block for block in self.snake.body[1:]
+```
+
+Como ya sabemos, esto es una forma de crear una colección ordenada de elementos. En este caso, cada elemento será `True` o `False`. Será verdadero (`True`) si la posición de la cabeza es igual a la tupla `block` (ya veremos de dónde sale `block`), y falso (`False`) en otro caso. La tupla `block` no es más que cada elemento del cuerpo de la serpiente (campo `body` del objeto `snake`). No obstante, recordemos que el primer elemento del cuerpo de la serpiente es la cabeza pero no nos interesa comprobar que la cabeza haya chocado consigo misma, por tanto solamente consideramos los elementos de esta lista a partir del segundo. Esto lo logramos con `self.snake.body[1:]`.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_6.md:237]: Maybe we should add a small reference to what is `slicing` in Python

File: tutorial/guias/Paso_a_paso_DIA_6.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+    )
+```
+
+Finalmente, la función `_generate_fruit`, genera una nueva fruta en una posición aleatoria. Esto lo logramos usando el módulo `random`, específicamente la función `randrange` de este módulo. Lo que queremos es generar una posición aleatoria para pasársela a la clase `Fruit`. También debemos recordar que nuestra fruta se representa con un cuadrado de 10 píxeles de lado. Por tanto debemos ubicarla de forma que quepa dentro de nuestra ventana de juego sabiendo el área que va a ocupar. La función `randrange` nos devuelve un número entero aleatorio que se encuentra en el rango definido por los parámetros que le pasamos. O sea `random.randrange(1, 10)` nos da un número aleatorio entre uno y diez. En este caso queremos un número que sea mayor que 0 y que sea menor que el ancho de la pantalla divido entre 10 para la posción horizontal (para la vertical es lo mismo pero en lugar del ancho es el alto lo que dividimos entre 10). Luego este número lo multiplicamos por 10 para obtener la coordenada donde representaremos la fruta. Con esto logramos que la coordenada de nuestra fruta se encuentre en números que son múltiplos de 10 y que no se salga de la pantalla de juego.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_6.md]: Typo: `posción` -> `posición`

File: tutorial/guias/Paso_a_paso_DIA_6.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+                self.paused = not self.paused
+            if event.key == pygame.K_q:
+                pygame.quit()
+                sys.exit()
+    return change_to
+```
+
+Aquí devolveremos el resultado de la interacción del usuario con el juego. Este resultado se almacenará en la variable `change_to` cuando implique un cambio de dirección. Inicialmente `change_to` contendrá la dirección actual de nuestra serpiente.
+
+Luego iteramos por todos los eventos que ha capturado Pygame. Durante la ejecución de nuestro juego, Pygame va almacenando todas las acciones que hacemos: teclas que presionamos, movimientos del mouse, clicks, etc. Estos eventos los podemos leer en nuestros programas y decidir qué hacer cuando ocurren. En nuestro caso, solamente nos interesará cuando el usuario presione las teclas de dirección para cambiar el movimiento de la serpiente, la tecla espaciadora para pausar el juego, o la tecla <kb>q</kb> para quitar el juego. El resto de eventos serán ignorados.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_6.md]: Change the keyboard tag to `<kbd></kbd>`. You will get something like this <kbd>q</kbd>.

File: tutorial/guias/Paso_a_paso_DIA_6.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+
+Los eventos, como ya vimos, pueden ser de varios tipos. El único tipo que nos interesa es el `KEYDOWN` que representa que se presionó una tecla. A partir de ahí podemos saber qué tecla se presionó a partir del campo `key` que poseen todos los eventos de tipo `KEYDOWN`. Pygame nos ofrece variables para referirnos a las teclas con nombres sugerentes. Por ejemplo, la flecha de dirección hacia arriba es `K_UP`. Si la tecla presionada representa una dirección, almacenamos la dirección correspondiente en la variable `change_to`. Si la tecla es la barra espaciadora, le asignamos al campo `paused` el valor contrario al que tenía antes (si era `True` lo cambiamos a `False` y viceversa), de esta forma pausamos y reanudamos el juego. Por último, si se presiona la tecla <kb>q</kb> cerramos el juego usando `pygame.quit()` y `sys.exit()`. Al final devolvemos la variable `change_to` con la dirección actualizada.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_6.md]: Same comment as above

File: tutorial/guias/Paso_a_paso_DIA_6.md [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+
+En caso de que no se haya comido una fruta, entonces tenemos que revertir el crecimiento de la serpiente y para eso usamos la función `trim`.
+
+Luego usamos la función `render_game_sate` de la clase `Renderer` para representar todos los elementos del juego. Después toca verificar si hemos perdido, o sea, si la función `game_over` nos devuelve `True` en las condiciones actuales del juego. En caso afirmativo, procedemos a dibujar la pantalla de *game over*. Usamos la fuente *time new roman* con tamaño 50. Dejamos esta pantalla durante 2 segundos con `time.sleep(2)` y luego cerramos el juego.
+
+Si no hemos perdido aún, representamos nuestro score en pantalla. Usaremos para ello la fuente *times new roman* pero esta vez con tamaño 20.
+
+Usamos la función `pygame.display.update()` para que nuestra pantalla se actualice con la situación actual del juego y adaptamos los *fraps per second* a la velocidad del juego con `self.fps.tick(self.snake_speed)`.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_6.md]: Again, I think it is `frames per second`, but now I'm starting to think it was not a typo on your side 😂. So, maybe you are right.

**Review by albexl — CHANGES_REQUESTED** [review]
I left some comments. After fixing them, we can merge for sure!

File: tutorial/guias/Paso_a_paso_DIA_6.md (hunk 1 above)
Comment (josejorgers) [reply tutorial/guias/Paso_a_paso_DIA_6.md]: Yes, I don't know where I got fraps from

File: tutorial/guias/Paso_a_paso_DIA_6.md (hunk 2 above)
Comment (josejorgers) [reply tutorial/guias/Paso_a_paso_DIA_6.md:237]: It could also be a new blog posts hehe

**Pull request merged**